File: Spring/src/main/java/com/nakul/socialmedia/demo/dao/UserDao.java
package com.nakul.socialmedia.demo.dao;

import com.nakul.socialmedia.demo.models.UserModel;
import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Component
public class UserDao {

    private SessionFactory sessionFactory;

    @Transactional
    public List<UserModel> getUsers() {
        Session session = sessionFactory.getCurrentSession();
        List<UserModel> users = session.createQuery("from user", UserModel.class).list();
        return users;
    }
}

File: UserDemo/src/main/java/com/nakul/user/dao/UserDao.kt
package com.nakul.user.dao

import com.nakul.user.model.User


interface UserDao {

    fun create(user: User): User?


    fun read(): List<User>?

    fun readById(id: Int): User?


    fun update(id: Int? = null, user: User? = null):User?

    fun delete(id: Int)
}

